### PR TITLE
dbeaver/pro#1272 use "int"/"integer" synonyms for the better data typ…

### PR DESCRIPTION
**Pull request by LonwoLonwo**
…es mapping

2022-11-22 20_13_57-143 453 106 - AnyDesk

Qriteries:

- [ ] The XLSX file can be imported in the new table with the "INT"/"INTEGER" data type for the new numeric column
- [ ] Tables with INT columns can be imported into other databases. INT data type will be not converted or converted to INTEGER.
- [ ] Tables with INTEGER columns can be imported into other databases. The INTEGER data type will be not converted or converted to INT.

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java [hunk 1]
```diff
@@ -351,41 +353,53 @@ public static String mapTargetDataType(DBSObject objectContainer, DBSTypedObject
                             }
                         }
                         if (targetType == null) {
-                            if (typeNameLower.contains("float") ||
-                                typeNameLower.contains("real") ||
+                            String intDataType = "int";
+                            String integerDataType = "integer";
+                            String floatDataType = "float";
+                            String realDataType = "real";
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java]: These are not being reassigned anywhere. Please make them `static final` constants.

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java [hunk 2]
```diff
@@ -317,7 +317,8 @@ public static String mapTargetDataType(DBSObject objectContainer, DBSTypedObject
                 // It seems this data type has modifiers. Try to find without modifiers
                 dataType = dataTypeProvider.getLocalDataType(SQLUtils.stripColumnTypeModifiers(typeName));
             }
-            if (dataType == null && typeNameLower.equals("double")) {
+            String doubleDataType = "double";
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java]: Same here

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java [hunk 3]
```diff
@@ -340,6 +341,7 @@ public static String mapTargetDataType(DBSObject objectContainer, DBSTypedObject
                 if (!possibleTypes.isEmpty()) {
                     // Try to get any partial match
                     targetType = possibleTypes.get(typeNameLower);
+                    String textDataType = "text";
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java]: Same here

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java (hunk 1 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java]: 2022-11-23 11_37_54-143 453 106 - AnyDesk

And I do not see any reason to move as Util class variables.

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java (hunk 1 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java]: It's about code smell. No one does that, everyone use class constants.

And the second reason is that local variable access is not optimized. Turning these into constants will make the compiler inline them without any runtime cost. Of course if wouldn't affect the performance that much, but anyways.

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java]: I will repeat it - Util class is not an appropriate place for the class constants. It is a commonplace. It has many not bounded methods.

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**